app/questionnaire/rules.py

```diff
@@ -210,15 +215,24 @@ def _get_when_rule_value(
     elif 'meta' in when_rule:
         value = get_metadata_value(metadata, when_rule['meta'])
     elif 'list' in when_rule:
-        value = get_list_count(list_store, when_rule['list'])
+        if 'id_selector' in when_rule:
```

Comment:
```Python
try:
    value = getattr(list_store.get(when_rule['list']), when_rule['id_selector'])
except KeyError:
    value = get_list_count(list_store, when_rule['list'])
```

Reply:
Neither one of these is an exception, where this change implies one of them is. I think the original is more reasonable.

Reply:
I disagree - the suggested change is an idiomatic approach to handling mapping access control flow.

Reply:
I'm sorry but I still maintain that my change is more readable and consistent with what is already there. The project hasn't ever prescribed the EAFP approach as far as I'm aware, so I'm not sure why this should block the PR?

Reply:
I don't really see much of a difference in readability, but the if pattern does keep this consistent with the rest of runner. 

We should probably avoid having this argument on all PRs until we decide on an approach with the whole team.